library/L2_HAL/communication/infrared_receiver.hpp

```diff
@@ -0,0 +1,37 @@
+#pragma once
+
+#include <cstdint>
+#include <functional>
+
+#include "utility/status.hpp"
+
+namespace sjsu
+{
+/// An abstract interface for communication drivers responsible for receiving
+/// infrared (IR) data.
+class InfraredReceiver
+{
+ public:
+  /// IR data frame containing the timestamps for the received de-modulated IR
+  /// signal.
+  struct DataFrame_t
+  {
+    /// Maximum size of the pulse_buffer.
+    inline static constexpr uint32_t kMaxPulseBufferSize = 100;
```

Comment:
```suggestion
    inline static constexpr uint32_t kMaxPulseBufferSize = 100;
```
The size for this is quite large. Does it need to be this large?

Reply:
Are the timestamps between the pulses so large that we need 32 bits of resolution or can 8bit or 16bit suffice?